tests/modules/apigee_mock_api_client/tests/src/Traits/ApigeeMockApiClientHelperTrait.php

```diff
@@ -0,0 +1,269 @@
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\Tests\apigee_mock_api_client\Traits;
+
+use Apigee\Edge\Api\Management\Entity\Company;
+use Apigee\Edge\Api\Management\Entity\Organization;
+use Apigee\MockClient\Generator\ApigeeSdkEntitySource;
+use Drupal\apigee_edge\Entity\Developer;
+use Drupal\apigee_edge\Entity\DeveloperInterface;
+use Drupal\apigee_mock_api_client\Plugin\KeyProvider\TestEnvironmentVariablesKeyProvider;
+use Drupal\key\Entity\Key;
+use Drupal\user\UserInterface;
+use Http\Message\RequestMatcher\RequestMatcher;
+
+/**
+ * Helper functions working with Apigee tests.
+ */
+trait ApigeeMockApiClientHelperTrait {
+
+  /**
+   * The SDK connector service.
+   *
+   * @var \Drupal\apigee_edge\SDKConnectorInterface
+   */
+  protected $sdkConnector;
+
+  /**
+   * The mock handler stack is responsible for serving queued api responses.
+   *
+   * @var \Drupal\apigee_mock_api_client\MockHandlerStack
+   */
+  protected $stack;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The mock response factory service.
+   *
+   * @var \Apigee\MockClient\ResponseFactoryInterface
+   */
+  protected $mockResponseFactory;
+
+  /**
+   * If integration (real API connection) is enabled.
+   *
+   * @var bool
+   */
+  protected $integration_enabled;
+
+  /**
+   * The Apigee Edge key used in tests.
+   *
+   * @var string
+   */
+  protected $apigee_edge_test_key = 'apigee_edge_test_auth';
+
+  /**
+   * Setup.
+   */
+  protected function apigeeTestHelperSetup() {
+    $this->apigeeTestPropertiesSetup();
+    $this->initAuth();
+  }
+
+  /**
+   * Setup.
+   */
+  protected function apigeeTestPropertiesSetup() {
+    $this->stack = $this->container->get('apigee_mock_api_client.mock_http_handler_stack');
+    $this->sdkConnector = $this->container->get('apigee_edge.sdk_connector');
+    $this->entityTypeManager = $this->container->get('entity_type.manager');
+    $this->mockResponseFactory = $this->container->get('apigee_mock_api_client.response_factory');
+    $this->integration_enabled = getenv('APIGEE_INTEGRATION_ENABLE');
+  }
+
+  /**
+   * Initialize SDK connector.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  protected function initAuth() {
+
+    // Create new Apigee Edge basic auth key.
+    $key = Key::create([
+      'id'           => $this->apigee_edge_test_key,
+      'label'        => 'Apigee M10n Test Authorization',
+      'key_type'     => 'apigee_auth',
+      'key_provider' => 'apigee_edge_environment_variables',
+      'key_input'    => 'apigee_auth_input',
+    ]);
+
+    $key->save();
+
+    // Collect credentials from environment variables.
+    $fields = [];
+    foreach (array_keys($key->getKeyType()->getPluginDefinition()['multivalue']['fields']) as $field) {
```

Comment:
How about not using the environment vars key provider, but instead setting a "fake" key that that has a key provider of type "config". The values are not used for anything, so why try to keep around the real values set from the environment? This way we don't have to worry about moving around env vars, setting those vars in state, etc.  We could remove `tests/modules/apigee_mock_api_client/src/Plugin/KeyProvider/TestEnvironmentVariablesKeyProvider.php` and maybe other code.

Reply:
I like this idea too, we can add a follow up task to implement this after some other parts of the tests refactoring are in place.